File: 3453-generate-binary-strings-without-adjacent-zeros/generate-binary-strings-without-adjacent-zeros.rs
impl Solution {
    pub fn valid_strings(n: i32) -> Vec<String> {
        let mut res = Vec::new();
        Self::backtrack(n as usize, String::new(), false, &mut res);
        res
    }
    fn backtrack(n: usize, curr: String, prev: bool, res: &mut Vec<String>) {
        if curr.len() == n {
            res.push(curr);
            return;
        }
        Self::backtrack(n, curr.clone() + "1", false, res);
        if !prev {
            Self::backtrack(n, curr + "0", true, res);
        }
    }
}

File: 3453-generate-binary-strings-without-adjacent-zeros/generate-binary-strings-without-adjacent-zeros.py
class Solution(object):
    def validStrings(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        def backtrack(curr, prev):
            if len(curr) == n:
                ans.append(curr)
                return
            backtrack(curr + '1', False)
            if not prev:
                backtrack(curr + '0', True)
        ans = []
        backtrack('', False)
        return ans